### PR TITLE
Docs / Vue - Getting Started - Examples

### PR DESCRIPTION
**Pull request by CyberStrike**
Issue:

## What I did
Update Vue Documentation With Examples for 
- Whats A Story / Vue Button Story With Args
- Setup / Vue - Your Component
- Setup / Vue - Preview Global Decorator

## How to test

- Is this testable with Jest or Chromatic screenshots? **No.**
- Does this need a new example in the kitchen sink apps? **No.**
- Does this need an update to the documentation? **Yes.**

If your answer is yes to any of these, please make sure to include it in your PR.

### CODE REVIEW

**Comment by jonniebigodes** [comment]
@CyberStrike i'll take a look at it in a bit. I want to check what snippets i have in place so that we don't overlap things. Do you mind waiting a bit while confirm this?

Stay safe

**Comment by CyberStrike** [comment]
> @CyberStrike i'll take a look at it in a bit. I want to check what snippets i have in place so that we don't overlap things. Do you mind waiting a bit while confirm this?
> 
> Stay safe

**Review by shilman — APPROVED** [review]
LGTM. is this still a draft?

**Comment by CyberStrike** [comment]
@shilman It's good to go from me, but I noticed a failing check is there something I need to fix on my end, Danger error wasn't clear.

File: docs/snippets/vue/storybook-preview-global-decorator.js.mdx [hunk 1]
```diff
@@ -1,5 +1,5 @@
 ```js
 // .storybook/preview.js
 
-export const decorators = [() => ({ template: '<div style="margin: 3em;"><story /></div>' })];
+export const decorators = [(story) => ({ template: '<div style="margin: 3em;"><story /></div>' })];
```
Comment (jonniebigodes) [inline docs/snippets/vue/storybook-preview-global-decorator.js.mdx:4]: @CyberStrike you can rename the file to: "storybook-preview-with-styled-components-decorator.js.mdx" and add the following:
```js
// .storybook/preview.js
import { ThemeProvider } from "vue-styled-components";


export const decorators = [
  (story) => ({
    components: {
      "theme-provider": ThemeProvider,
    },
    template: `<theme-provider :theme="{
    primary: 'palevioletred',
    secondary: 'papayawhip',
    tertiary: 'lavenderblush'
  }"><story/></theme-provider>`,
  }),
];
```

I did a little bit of investigation and it seems that styled components are available for Vue. Which applied to this case we can maintain the same pattern. Pardon the bad indent.

**Review by jonniebigodes — CHANGES_REQUESTED** [review]
@CyberStrike overall looks good to me. Really appreciate the help with this, when you're able adjust take a look at the comment i left and we'll be good to merge when you move it from draft status.

Let me know and we'll go from there.

Stay safe

File: docs/snippets/vue/storybook-preview-global-decorator.js.mdx (hunk 1 above)
Comment (CyberStrike) [reply docs/snippets/vue/storybook-preview-global-decorator.js.mdx:4]: Got it will update.

Comment (CyberStrike) [reply docs/snippets/vue/storybook-preview-global-decorator.js.mdx:4]: `storybook-preview-global-decorator.js.mdx` is being used on `writing-stories/decorators` so I've kept it and created a new `storybook-preview-with-styled-components-decorator.js.mdx`

A note on this section for Vue developers chances are they would run into issues with Vue Modules when setting up an existing project at this point. But additions is another scope.

**Comment by jonniebigodes** [comment]
@CyberStrike sorry for not getting this merged sooner, i've checked and it all seems ok on my end from a snippets point of view. There were some conflicts that needed to be fixed as a couple of pull requests were merged earlier and i took care of them, if @shilman doesn't get this merged in between i'll take care of it tomorrow or the following day.

**Comment by NickMcBurney** [comment]
@jonniebigodes do you know why docs are not updated following this PR being merged? I've been working with Vue decorators and was going to add PR for the [styled components](https://github.com/storybookjs/storybook/pull/13498/files#diff-38cb9f32718e167728bd3b46997cca1ce6382f80ef849548373d648855ddbb52), but I can see its already done here back in January. but the docs website still shows "This snippet doesn't exist for Vue yet..." https://storybook.js.org/docs/vue/writing-stories/decorators#context-for-mocking

**Pull request merged**